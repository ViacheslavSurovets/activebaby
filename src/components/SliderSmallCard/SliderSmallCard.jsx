import React from 'react';
import PropTypes from 'prop-types';
import {
  Slide,
  SliderSmallCardText,
  SliderSmallImage,
  SliderSmallLabel,
  SliderSmallLabelText
} from './styles';
import { useTranslation } from 'react-i18next';


const SliderSmallCard = ( props ) => {
  const { t } = useTranslation ();
  const { imageUrl, title, label, path } = props;

  return (
    <Slide to={path} className="swiper-slide" url={ imageUrl }>
      <SliderSmallImage src={ imageUrl } alr={ title } />
      <SliderSmallCardText>{ title }</SliderSmallCardText>
      <SliderSmallLabel>
        <SliderSmallLabelText>{ t(label) }</SliderSmallLabelText>
      </SliderSmallLabel>
    </Slide>
  );
};

export default SliderSmallCard;

SliderSmallCard.propTypes = {
  imageUrl: PropTypes.any,
  title: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired
};


