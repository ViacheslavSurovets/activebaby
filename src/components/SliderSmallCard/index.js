export { default as SliderSmallCard } from './SliderSmallCard';
