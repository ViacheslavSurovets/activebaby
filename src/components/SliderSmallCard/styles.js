import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { flexCommon } from '@src/styles/commonStylesAndStyledComponents';

export const Slide = styled ( Link )`
  display: flex;
  flex-direction: column;
  width: 30vw;
  height: 20vw;
  text-decoration: none;
  position: relative;
`;

export const SliderSmallImage = styled.img`
  width: 100%;
  height: 80%;
`;

export const SliderSmallCardText = styled.span`
  ${ flexCommon };
  height: 20%;
  font-size: 1.6rem;
  color: ${ ( { theme } ) => theme.main.sliderSmallCard.sliderSmallCardText }
`;

export const SliderSmallLabel = styled.div`
  ${flexCommon};
   position: absolute;
   top:0;
   right: 0; 
   background:tomato; 
`;

export const SliderSmallLabelText = styled.span`
  color: white;
  font-size: 2vw;
  padding: .5vw 1vw;
`;
