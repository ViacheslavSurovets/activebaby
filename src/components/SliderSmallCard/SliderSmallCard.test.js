import React from 'react';
import { shallow } from 'enzyme';
import SliderSmallCard from './SliderSmallCard';


describe ( 'SliderSmallCard', () => {

   const [imageUrl, title, label, path] = 'string';

  it ( 'SliderSmallCard Snapshot', () => {
    expect ( shallow (
      <SliderSmallCard
        imageUrl={imageUrl}
        title={title}
        label={label}
        path={path}
      /> ) ).toMatchSnapshot ();
  } );
} );
