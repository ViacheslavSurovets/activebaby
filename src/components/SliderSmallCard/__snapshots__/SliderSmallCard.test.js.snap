// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderSmallCard SliderSmallCard Snapshot 1`] = `ShallowWrapper {}`;
